ENABLE_AVX2=OFF doesn't work for CMake

**Describe the bug**
The cmake build of FLINT 3.2.0 fails for the conda-forge distribution on Windows.

The full log is [here](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1206745&view=logs&jobId=fe7faf59-69bb-5487-b8fd-a35a7bda3a51&j=fe7faf59-69bb-5487-b8fd-a35a7bda3a51&t=55731638-dd49-5144-2129-5101b9dd1960) , the relevant error message is:
```
[1/5074] Building C object CMakeFiles\flint.dir\src\mpn_extras\mul.c.obj
FAILED: CMakeFiles/flint.dir/src/mpn_extras/mul.c.obj 
C:\PROGRA~1\MICROS~2\2022\ENTERP~1\VC\Tools\MSVC\1443~1.348\bin\Hostx64\x64\cl.exe  /nologo -DFLINT_BUILD_DLL -Dflint_EXPORTS -I%SRC_DIR%\src -I%SRC_DIR%\build -external:I%PREFIX%\Library\include -external:W0 /experimental:c11atomics /std:c11  /DWIN32 /D_WINDOWS /O2 /Ob2 /DNDEBUG -std:c11 -MD /experimental:c11atomics /showIncludes /FoCMakeFiles\flint.dir\src\mpn_extras\mul.c.obj /FdCMakeFiles\flint.dir\ /FS -c %SRC_DIR%\src\mpn_extras\mul.c
%SRC_DIR%\src\machine_vectors.h(1631): fatal error C1189: #error:  machine_vector.h not implemented
```

**Steps to reproduce**

We use this script to build https://github.com/conda-forge/libflint-feedstock/blob/main/recipe/bld.bat.

I don't have a more minimal reproducer at the moment but could try to produce one (I don't have Windows myself, so I am not sure what you would need.)

**Expected behavior**
Build should pass without errors.

**System (please complete the following information):**
* Your OS and its version: Microsoft Windows Server 2022
* The CPU you are on: not sure. The one provided by Azure Cloud.
* The version of FLINT: 3.2.0 from the tarball
* How you configured FLINT:
```
cmake ^
  -G "Ninja" ^
  -DBUILD_TESTING=ON ^
  -DCMAKE_BUILD_TYPE=Release ^
  -DCMAKE_INSTALL_PREFIX=%LIBRARY_PREFIX% ^
  -DCMAKE_PREFIX_PATH=%LIBRARY_PREFIX% ^
  -DENABLE_AVX2=OFF ^
  -DENABLE_AVX512=OFF ^
  ..
```

See https://github.com/conda-forge/libflint-feedstock/pull/40 for the attempt to package FLINT in the conda-forge distribution.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -150,7 +150,7 @@ set(CMAKE_REQUIRED_LIBRARIES gmp::gmp)
 if(HAS_FLAG_ARCH)
     set(CMAKE_REQUIRED_FLAGS "-march=${ENABLE_ARCH}")
 endif()
-if(HAS_FLAG_AVX2)
+if(ENABLE_AVX2)
     set(CMAKE_REQUIRED_FLAGS "${avx2_flag}")
 endif()
 check_c_source_compiles([[

